### PR TITLE
Refactor data and main render loop

### PR DESCRIPTION
**Pull request by swemoney**
This refactors the mlbgame data into a separate object that can be passed around. Also refactors the main renderer to use a single loop. Everything should still work as it did before but the refactor cleaned up the random flashing issues and should help clean things up a bit. Needs to be tested by more people but should be ready for the dev branch.

### CODE REVIEW

File: data/data.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+NETWORK_RETRY_SLEEP_TIME = 5.0
+
+class Data:
+  def __init__(self, config):
+    # Save the parsed config
+    self.config = config
+
+    # Parse today's date and see if we should use today or yesterday
+    self.year, self.month, self.day = self.__parse_today()
+    # self.year, self.month, self.day = (2018, 4, 22)
```
Comment (ajbowler) [inline data/data.py]: remove commented line

File: data/data.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+    # Fetch all standings data for today
+    # (Good to have in case we add a standings screen while rotating scores)
+    self.refresh_standings()
+
+    # What game do we want to start on?
+    self.current_game_index = self.game_index_for_preferred_team()
+
+
+  #
```
Comment (ajbowler) [inline data/data.py]: Could you clean up this area there's some extra whitespace and some unused comments

File: data/data.py [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    end_of_day = datetime.strptime(self.config.end_of_day, "%H:%M").replace(year=today.year, month=today.month, day=today.day)
+    if end_of_day > datetime.now():
+      today -= timedelta(days=1)
+    return (today.year, today.month, today.day)
+
+  def date(self):
+    return datetime(self.year, self.month, self.day)
+
+
+  #
```
Comment (ajbowler) [inline data/data.py]: clean up

File: main.py [hunk 4]
```diff
@@ -1,9 +1,11 @@
 from datetime import datetime, timedelta
 from data.scoreboard_config import ScoreboardConfig
-from renderers.games import GameRenderer
+# from renderers.games import GameRenderer
```
Comment (ajbowler) [inline main.py]: remove commented line

File: renderers/games.py [hunk 5]
```diff
@@ -61,23 +58,22 @@ def render(self):
     is required.
     """
     self.canvas.Fill(*ledcolors.scoreboard.fill)
-    current_game_index = self.__get_game_from_args()
-    game = self.games[current_game_index]
+    # current_game_index = self.data.game_index_for_preferred_team()
```
Comment (ajbowler) [inline renderers/games.py]: remove commented lines

File: renderers/main.py [hunk 6]
```diff
@@ -0,0 +1,133 @@
+from data.final import Final
```
Comment (ajbowler) [inline renderers/main.py]: Some of these imports definitely aren't being used (mlbgame at least), could you remove?

File: renderers/main.py [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+      # If we're not scrolling anything, scroll is always finished.
+      if Status.is_static(self.data.overview.status):
+        self.scrolling_finished = True
+
+      time.sleep(refresh_rate)
+      endtime = time.time()
+      time_delta = endtime - starttime
+
+      # Determine the rate in which we're rotating to the next screen
+      rotate_rate = self.data.config.live_rotate_rate
```
Comment (ajbowler) [inline renderers/main.py]: Could you move this block out to its own method?

File: utils.py [hunk 8]
```diff
@@ -17,11 +17,11 @@ def center_text_position(text, canvas_width):
 def split_string(string, num_chars):
   return [(string[i:i + num_chars]).strip() for i in range(0, len(string), num_chars)]
 
-def bump_counter(counter, arr):
-  counter += 1
-  if counter >= len(arr):
-    counter = 0
-  return counter
+# def bump_counter(counter, arr):
```
Comment (ajbowler) [inline utils.py]: Delete

File: renderers/main.py (hunk 6 above)
Comment (swemoney) [reply renderers/main.py]: I removed a bunch of imports. The only one I wasn't 100% sure about was math but I wasn't able to find anywhere where we were using it so hopefully nothing breaks down.

File: renderers/games.py (hunk 5 above)
Comment (swemoney) [reply renderers/games.py]: I went ahead and removed this comment but, ideally we don't need this file at all. It's been replaced with the MainRenderer. Was just keeping it around until we were sure everything worked.

**Comment by swemoney** [comment]
Requests have been taken care of. I'm a little curious about an issue I noticed last night where the scoreboard just stopped updating at about 6:30pm. Checked the logs and the last printed update was at that time and the scoreboard was stuck on a SEA/CLE game in the 5th inning all night until restarted.

**Comment by swemoney** [comment]
Nevermind. I'm an idiot for not noticing the status in the logs just before it stopped updating. Added new status, `Review` which should fix the issue.

**Pull request merged**